### PR TITLE
Add some more tests for logs

### PR DESCRIPTION
**Pull request by jrockway**
This revises how we interact with Loki.  To translate TimeRange filters into a set of Loki requests, we now have a TimeIterator object that spits out the times necessary to keep Loki happy, and a BatchedQueryRange function that executes queries against all the time ranges.  As a side effect, we get paging information out of the Time Iterator that is aware of limits / filters / offsets / etc.

All Loki configuration needed to make things work is queried from Loki (max batch size, max range).  This will allow people to change those settings on Loki without breaking everything.

I added some test that you can iterate over a set of logs in forward and reverse with the paging hints.

### CODE REVIEW

File: src/logs/logs.proto [hunk 1]
```diff
@@ -79,20 +79,30 @@ message JobDatumLogQuery {
...
 message LogFilter {
+  // If set, only return logs in the provided time range.
   TimeRangeLogFilter time_range = 1;
+  // If set, return at maximum this number of logs.
   uint64 limit = 2;
+  // If set, only return logs that match this regular expression.
   RegexLogFilter regex = 3;
-  // Minimum log level to return; worker will always run at level debug, but setting INFO here restores original behavior
+  // If set, only return logs that are greater than or equal to this log level.  (DEBUG returns
+  // DEBUG, INFO, ERROR, INFO return s INFO and ERROR, etc.).
```
Comment (bbonenfant) [inline src/logs/logs.proto:91]: `return s` -> `returns`

**Review by bbonenfant — APPROVED** [review]
🎉 More proto comments 🎉

**Review by acohen4 — APPROVED** [review]
LGTM!

File: src/logs/logs.proto (hunk 1 above)
Comment (jrockway) [reply src/logs/logs.proto]: Done.

**Pull request merged**